components/async-container/demo/async-item.js

```diff
@@ -0,0 +1,66 @@
+import '../../colors/colors.js';
+import { css, html, LitElement } from 'lit-element/lit-element.js';
+import { InitialStateError, runAsync } from '../../../directives/run-async.js';
+
+class AsyncItem extends LitElement {
+
+	static get properties() {
+		return {
+			delay: { type: Number },
+			key: { type: String },
+			throwError: { type: Boolean, attribute: 'throw-error' }
+		};
+	}
+
+	static get styles() {
+		return css`
+			:host {
+				display: inline-block;
+			}
+			div {
+				background-color: var(--d2l-color-celestine);
+				border: 1px solid var(--d2l-color-galena);
+				border-radius: 0.4rem;
+				color: white;
+				padding: 0.3rem;
+				width: 100px;
+				height: 100px;
+				text-align: center;
+			}
+		`;
+	}
+
+	constructor() {
+		super();
+		this.delay = 4000;
+		this.key = null;
+	}
+
+	render() {
+		return html`${runAsync(this.key, (key) => this._getContent(key), {
```

Comment:
I used `runAsync` directive here simply for simplicity in this demo component.  I would probably provide another demo component that shows how to dispatch the event more directly...
```
element.dispatchEvent(new CustomEvent('pending-state', {
		composed: true,
		bubbles: true,
		detail: { promise: pendingPromise }
}));
```